java/core/src/main/java/com/whylogs/core/metrics/Metric.java

```diff
@@ -0,0 +1,48 @@
+package com.whylogs.core.metrics;
+
+import com.whylogs.core.PreProcessedColumn;
+import com.whylogs.core.SummaryConfig;
+
+import java.util.HashMap;
+
+public abstract class Metric{
+
+    private String namespace;
+
+    // TODO: needs a __post_init__ for the registry
+
+    public static String getNameSpace(Metric metric, MetricConfig config){
+        return metric.zero(config).namespace;
+    }
+
+    public abstract <T extends Metric> T zero(MetricConfig config);
+
+    public <T extends Metric> T add(Class<T> otherMetric){
+        return this.merge(otherMetric);
+    }
+
+    public <T extends Metric> T merge(Class<T> otherMetric){
+        // TODO: Metric Components
+        return null;
+    }
+
+    // TODO: protobuf needs MetricMessage
+
+    // TODO: get_component_paths needs MetricComponents
+
+    public abstract <O> HashMap<String, O> toSummaryDict(SummaryConfig config); // TODO: this doesn't make a good api so ...
```

Comment:
I think we want `HashMap<String, Object>` ?

Reply:
Yes :) That was me playing with something and forgot to switch it back.